### PR TITLE
HSEARCH-2955 Faceting query fails when the facet field is not populated for any document

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-2955

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/query/engine/impl/QueryHits.java [hunk 1]
```diff
@@ -399,13 +399,20 @@ private FacetResult getFacetResultForLongRange(RangeFacetRequest<?> facetRequest
...
 		SortedSetDocValuesReaderState docValuesReaderState;
 		try {
 			docValuesReaderState = new DefaultSortedSetDocValuesReaderState( searcher.getIndexReader() );
 		}
 		catch (IllegalArgumentException e) {
-			// happens in case there are no facets at all configured for the matching documents
-			throw log.unknownFieldNameForFaceting( facetRequest.getFacetingName(), facetRequest.getFieldName() );
+			/*
+			 * Happens in case there are no facets at all stored in the matching documents.
+			 * But we know some the target field is correctly configured to generate facets,
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/search/query/engine/impl/QueryHits.java]: s/some// ?

File: engine/src/main/java/org/hibernate/search/query/engine/impl/QueryHits.java [hunk 2]
```diff
@@ -399,13 +399,20 @@ private FacetResult getFacetResultForLongRange(RangeFacetRequest<?> facetRequest
...
+			 * because we managed to retrieve the FacetMetadata.
+			 * So we can safely return an empty list: the matching documents simply do not have
+			 * any value for this field.
+			 */
+			return facets;
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/search/query/engine/impl/QueryHits.java]: Can't we use `Collections.emptyList()` and keep the array list initialization where it was? I think the list is readonly after that? Let's not initialize an array list for the sake of it.

Comment (yrodiere) [reply engine/src/main/java/org/hibernate/search/query/engine/impl/QueryHits.java]: Well the facet list we return to the user has always been mutable, and I'm dead certain some people must rely on it.
Plus the calling code mutates the list in some cases. Not when it's empty, granted, but still I would feel bad returning an immutable list.

So... Not in a minor?

File: engine/src/main/java/org/hibernate/search/query/engine/impl/QueryHits.java (hunk 1 above)
Comment (yrodiere) [reply engine/src/main/java/org/hibernate/search/query/engine/impl/QueryHits.java]: Right, done.

File: engine/src/main/java/org/hibernate/search/query/engine/impl/QueryHits.java (hunk 2 above)
Comment (gsmet) [reply engine/src/main/java/org/hibernate/search/query/engine/impl/QueryHits.java]: Right, I thought we would have made them immutable at some point.

Merged!

**Pull request merged**